drivers/edac/shell.c

```diff
@@ -58,10 +58,47 @@ static void decode_ecc_error(const struct shell *shell, uint64_t ecc_error)
 	}
 }
 
+static int shell_ecc_error_show(const struct shell *sh,
+				const struct device *dev)
+{
+	uint64_t error;
+	int err;
+
+	err = edac_ecc_error_log_get(dev, &error);
+	if (err != 0 && err != -ENODATA) {
+		shell_error(sh, "Error getting error log (err %d)", err);
+		return err;
+	}
+
+	shell_fprintf(sh, SHELL_NORMAL, "ECC Error: 0x%llx\n", error);
+
+	if (error != 0) {
+		decode_ecc_error(sh, error);
+	}
+
+	return 0;
+}
+
+static int shell_parity_error_show(const struct shell *sh,
```

Comment:
Even though these are static functions, I think it’d be good to avoid violating the shell subsystem namespace, so please try to use some other prefix.

Reply:
renamed